src/DB/Table/BudgetRecommendationTable.php

```diff
@@ -14,6 +14,13 @@
  */
 class BudgetRecommendationTable extends Table {
 
+	/**
+	 * Whether the initial data has been loaded
+	 *
+	 * @var bool
+	 */
+	public $has_loaded_initial_data = false;
```

Comment:
💅  This could be private as far as I know

Reply:
Ok, I see its maybe for the tests

Reply:
To consider a getter/setter

Reply:
Yes, and I thought that it could be handy to allow the instance to know/access if the data was loaded.